executor/src/witgen/jit/function_cache.rs

```diff
@@ -89,9 +92,29 @@ impl<'a, T: FieldElement> FunctionCache<'a, T> {
         cache_key: &CacheKey,
     ) -> Option<WitgenFunction<T>> {
         log::trace!("Compiling JIT function for {:?}", cache_key);
+
         self.processor
             .generate_code(mutable_state, cache_key.identity_id, &cache_key.known_args)
             .ok()
+            .and_then(|code| {
+                // TODO: Remove this once BlockMachine passes the right amount of context for machines with
+                // non-rectangular block shapes.
+                let is_rectangular = code
+                    .iter()
+                    .filter_map(|effect| match effect {
+                        Effect::Assignment(v, _) => Some(v),
```

Comment:
Will this code be removed completely? I'm asking because `Assignment` is not the only effect that can perform an assignment.